### DISCUSSION TITLE
Select binary

### DISCUSSION THREAD

**Discussion by AstraLuma** [casey/just#1596] [General]
How should I select the right command to call, if different installations have different invocations?

I'm specifically looking at `podman-compose`, `docker-compose`, and `docker compose`, and which one to call is going to vary.

How do I check what programs are installed and select one?

**Comment by runeimp** [comment]
Is `docker-compose` ever the only choice without `docker compose` being an option? Commands that support subcommands often allow for external subcommands by prefixing the subcommand name with the command name ala `commandname-subcommand` so calling `commandname subcommand` checks for subcommand as an internal function and if that isn't found checks PATH for `commandname-subcommand`.

To answer your question; shell scripting of some sort? Something like the following...

```just
compose *args:
	#!/bin/sh

	if [ "$(which podman-compose; echo $?)" -eq 0 ]; then
		podman-compose podman-specific-argument {{args}}
	elif [ "$(which docker-compose; echo $?)" -eq 0 ]; then
		docker-compose docker-specific-argument {{args}}
	elif [ "$(which docker; echo $?)" -eq 0 ]; then
		docker compose docker-specific-argument {{args}}
	else
		echo "Sorry no composability commands found"
	fi
```

Or maybe using Nu or whatever cross-platform shell you use.

**Reply by AstraLuma** [reply]
`docker compose` isn't valid for older versions (I'd have to look when it was added).

**Reply by AstraLuma** [reply]
That sample is ok for one recipe, but it gets annoying if you have 3 or 5 recipes that all do compose-based manipulations.
